katalyze/main.go

```diff
@@ -70,80 +70,77 @@ func main() {
 		glog.Exit("--config=<analysis-config> must be specified, but was empty")
 	}
 
-	files := getSGFs(flag.Args())
+	files, err := filterSGFs(flag.Args())
+	if err != nil {
+		glog.Exit(err)
+	}
 	if len(files) == 0 {
 		glog.Exit("No SGF files specified -- must be specified as args to katalyze")
 	}
 
 	an := katago.New(model, analysisConfig, *analysisThreads)
-
-	if err := an.Start(); err != nil {
+	if err = an.Start(); err != nil {
 		glog.Exitf("error booting Katago: %v", err)
 	}
 
-	if err := process(files, an); err != nil {
-		glog.Exitf("error processing files: %v", err)
+	cwd, err := os.Getwd()
+	if err != nil {
+		glog.Exit(err)
 	}
+	outDir := cwd
+	if *outputDir != "" {
```

Comment:
Seems like it would make sense to use the DiskStorage here - but maybe you are trying to keep it decoupled from that.

Reply:
Right now disk-storage doesn't create the root directory, which is probably correct. So, I create it here if it doesn't exist. It does duplicate a bit what DiskStorage is doing, but it's probably the right separation of responsibilities.